manim/utils/hashing.py

```diff
@@ -0,0 +1,191 @@
+        except TypeError:
+            # This is used when the user enters an unknown type in CONFIG. Rather than throwing an error, we transform
+            # it into a string "Unsupported type for hashing" so that it won't affect the hash.
+            return "Unsupported type for hashing"
+
+    def _encode_dict(self, obj):
+        """Clean dicts to be serialized : As dict keys must be of the type (str, int, float, bool), we have to change them when they are not of the right type.
+        To do that, if one is not of the good type we turn it into its hash using the same
+        method as all the objects here.
+
+        Parameters
+        ----------
+        obj : :class:`any`
+            The obj to be cleaned.
+        
+        Returns
+        -------
+        `any`
+            The object cleaned following the processus above.
+        """
+
+        def key_to_hash(key):
+            if not isinstance(key, (str, int, float, bool)) and key is not None:
+                # print('called')
+                return zlib.crc32(json.dumps(key, cls=CustomEncoder).encode())
+            return key
+
+        if isinstance(obj, dict):
+            return {key_to_hash(k): self._encode_dict(v) for k, v in obj.items()}
+        return obj
+
+    def encode(self, obj):
+        return super().encode(self._encode_dict(obj))
+
+
+def get_json(obj):
+    """Recursively serialize `object` to JSON using the :class:`CustomEncoder` class.
+
+    Paramaters
+    ----------
+    dict_config : :class:`dict`
+        The dict to flatten
+
+    Returns
+    -------
+    :class:`str` 
+        The flattened object
+    """
+    return json.dumps(obj, cls=CustomEncoder)
+
+
+def get_camera_dict_for_hashing(camera_object):
+    """Remove some keys from `camera_object.__dict__` that are very heavy and useless for the caching functionality.
+
+    Parameters
+    ----------
+    camera_object : :class:`~.Camera`
+        The camera object used in the scene
+
+    Returns
+    -------
+    :class:`dict`
+        `Camera.__dict__` but cleaned.
+    """
+    camera_object_dict = copy.copy(camera_object.__dict__)
+    # We have to clean a little bit of camera_dict, as pixel_array and background are two very big numpy arrays.
+    # They are not essential to caching process.
+    # We also have to remove pixel_array_to_cairo_context as it contains used memory adress (set randomly). See l.516 get_cached_cairo_context in camera.py
+    for to_clean in ["background", "pixel_array", "pixel_array_to_cairo_context"]:
+        camera_object_dict.pop(to_clean, None)
+    return camera_object_dict
+
+
+def get_hash_from_play_call(camera_object, animations_list, current_mobjects_list):
+    """Take the list of animations and a list of mobjects and output their hashes. This is meant to be used for `scene.play` function.
+
+    Parameters
+    -----------
+    camera_object : :class:`~.Camera`
+        The camera object used in the scene.
+
+    animations_list : :class:`list`
```

Comment:
list of what? and we only need to be able to iterate over it (we can specify a tuple for example), so Iterable
```suggestion
    animations_list : Iterable[:class:`~.Animation`]
```